Use finally insead of isset on error

BackEndTea opened the pull request:
By using the `finally` keyword we don't need to check if the error 
was thrown, and then throw it again after the checks.

This _should_ also help a bit more in case one needs to debug where
an error came from, as the original error is still raised, instead of 
being caught and thrown again.

localheinz reviewed (approved):
Using `finally` without `catch` is ugly, in my opinion, but it works. 

👍

BackEndTea commented:
I think its more clear than the current solution. 
Plus, this is what finally is intended for. Code blocks that need to be executed regardless of whether an error is thrown.

Could you explain why it is ugly? Because to me it doesn't seem ugly, so i'd like to hear your idea on it.

BackEndTea committed:
Use finally insead of isset on error

By using the `finally` keyword we don't need to check if the error
was thrown, and then throw it again after the checks.

This _should_ also help a bit more in case one needs to debug where
an error came from, as the original error is still raised, instead of
being caught and thrown again.

localheinz commented:
@BackEndTea 

Only came across a usage of `try`-`finally` (without the `catch`) in a closed-source project a couple of months ago, and I thought it was weird to `try` something without `catch`ing any exceptions that might be thrown. 

Here it makes perfect sense.

👍

BackEndTea commented:
Thank you @localheinz, @asgrim & @sebastianbergmann
